### PR TITLE
Adding Microsoft.AspNetCore.EnsureJsonTrimmability host configuration

### PR DESCRIPTION
**Pull request by brunolins16**
Contributes https://github.com/dotnet/aspnetcore/issues/45527

### CODE REVIEW

File: src/WebSdk/ProjectSystem/Targets/Microsoft.NET.Sdk.Web.ProjectSystem.targets [hunk 1]
```diff
@@ -46,6 +46,12 @@ Copyright (c) .NET Foundation. All rights reserved.
     <EnableUnsafeBinaryFormatterSerialization Condition="'$(EnableUnsafeBinaryFormatterSerialization)' == ''">false</EnableUnsafeBinaryFormatterSerialization>
   </PropertyGroup>
 
+
+  <PropertyGroup Condition="'$(PublishTrimmed)' == 'true' Or '$(IsTrimmable)' == 'true' Or '$(PublishAot)' == 'true'">
```
Comment (eerhardt) [inline src/WebSdk/ProjectSystem/Targets/Microsoft.NET.Sdk.Web.ProjectSystem.targets:50]: `IsTrimmable` is more for libraries. Apps don't ever set this property. I think we should only default this value when `PublishTrimmed` or `PublishAot` == true.

File: src/Tests/Microsoft.NET.Sdk.Web.Tests/Properties/launchSettings.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (eerhardt) [inline src/Tests/Microsoft.NET.Sdk.Web.Tests/Properties/launchSettings.json:1]: Is this file necessary?

File: src/Tests/Microsoft.NET.Sdk.Web.Tests/PublishTests.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+namespace Microsoft.NET.Sdk.Web.Tests
+{
+    public class PublishTests : SdkTest
+    {
+        public PublishTests(ITestOutputHelper log) : base(log)
+        {
+        }
+
+        [RequiresMSBuildVersionTheory("17.0.0.32901")]
```
Comment (captainsafia) [inline src/Tests/Microsoft.NET.Sdk.Web.Tests/PublishTests.cs:20]: Why is this particular MSBuild version needed?

Comment (brunolins16) [reply src/Tests/Microsoft.NET.Sdk.Web.Tests/PublishTests.cs:20]: I don't believe it is needed, i copy/paste from a different test and might forgot to clean up. Let me check and fix it.

File: src/WebSdk/ProjectSystem/Targets/Microsoft.NET.Sdk.Web.ProjectSystem.props [hunk 4]
```diff
@@ -27,5 +27,4 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <Import Project="$(MSBuildThisFileDirectory)Microsoft.NET.Sdk.Web.DefaultItems.props"
     Condition="Exists('$(MSBuildThisFileDirectory)Microsoft.NET.Sdk.Web.DefaultItems.props')"/>
-
```
Comment (eerhardt) [inline src/WebSdk/ProjectSystem/Targets/Microsoft.NET.Sdk.Web.ProjectSystem.props:30]: Is this change necessary? Merging it would cause unnecessary churn in source control history.

File: src/Tests/Microsoft.NET.Sdk.Web.Tests/PublishTests.cs [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            responseFileContents.Should().Contain("--feature:Microsoft.AspNetCore.EnsureJsonTrimmability=true");
+        }
+
+        public static IEnumerable<object[]> SupportedTfms { get; } = new List<object[]>
+        {
+#if NET8_0
+            new object[] { ToolsetInfo.CurrentTargetFramework }
+#else
+#error If building for a newer TFM, please update the values above
+#endif
```
Comment (eerhardt) [inline src/Tests/Microsoft.NET.Sdk.Web.Tests/PublishTests.cs:73]: Is this the typical way we add a `TODO` when a new TFM gets added to add support for previous TFMs?

File: src/Tests/Microsoft.NET.Sdk.Web.Tests/PublishTests.cs [hunk 6]
```diff
@@ -0,0 +1,100 @@
+﻿using System.Collections.Generic;
```
Comment (eerhardt) [inline src/Tests/Microsoft.NET.Sdk.Web.Tests/PublishTests.cs:1]: needs a copyright

**Review by eerhardt — APPROVED** [review]

File: src/Tests/Microsoft.NET.Sdk.Web.Tests/PublishTests.cs (hunk 5 above)
Comment (brunolins16) [reply src/Tests/Microsoft.NET.Sdk.Web.Tests/PublishTests.cs:73]: I borrow this from https://github.com/dotnet/sdk/blob/7a05724a3a1aaf0c0f97f098ae174e3faf2589aa/src/Tests/Microsoft.NET.Publish.Tests/PublishTestUtils.cs#L10
